torch/_dynamo/variables/tensor.py

```diff
@@ -681,36 +684,59 @@ class TensorWithTFOverrideVariable(VariableTracker):
     def create(
         tx,
         tensor_variable,
-        orig_tensor_variable_source,
+        tensor_variable_source,
         torch_function_fn,
         subclass_type,
         **kwargs,
     ):
         var = TensorWithTFOverrideVariable(
             tensor_variable,
-            orig_tensor_variable_source,
+            tensor_variable_source,
             torch_function_fn,
             subclass_type,
             **kwargs,
         )
         # stash the subclass type to rewrap an output tensor if needed
-        tx.output.install_global(var.global_class_name(), subclass_type)
+        if var.global_class_name() not in tx.output.global_scope:
+            tx.output.install_global(var.global_class_name(), subclass_type)
+
         return var
 
     def __init__(
         self,
         tensor_variable,
-        orig_tensor_variable_source,
-        subclass_torch_function__func,
+        tensor_variable_source,
+        torch_function_fn,
         subclass_type,
         **kwargs,
     ):
         super().__init__(**kwargs)
         self.tensor_variable = tensor_variable
-        self.orig_tensor_variable_source = orig_tensor_variable_source
-        self.subclass_torch_function__func = subclass_torch_function__func
+        self.tensor_variable_source = tensor_variable_source
+        self.torch_function_fn = torch_function_fn
         self.subclass_type = subclass_type
 
+    def as_proxy(self):
+        return self.tensor_variable.as_proxy()
```

Comment:
I think I'd probably prefer NOT to define this method, as a tensor subclass is not proxyable in the naive way: if you have `tensor_subclass.add(2)`, it would be incorrect to trace this into the graph as `tensor_variable.add(2)` because this would bypass the torch function handling. Better to force end users to explicitly ask for the inner tensor variable proxy